Dark mode

TurturOk opened the pull request:
I have added **DarkModeSwitcher** component and placed it on top of every page. 
Mode preferrence is saved in local storage under "dark_mode" key. If enabled, "style_dark_mode" class is added to the body and styling rules are applied.
For convenience, I have replaced index.css with index.scss and added required dependencies. Is this acceptable?

Known issues:
- There is a flash of white when reloading the page. Rendering order should be somehow adjusted to prevent it.
- SE .png logo has dark text, that is hard to read in dark mode. Imho not a big problem, but might be good idea to provide alternative .png logo with white text and switch to it in dark mode.
- [EDIT] there is no tooltip for what the button does, did not know how translation works here

TurturOk committed:
Dark mode in progress: switcher component added, setting saved to local storage.

TurturOk committed:
fixed whatever prettier needed

TurturOk committed:
Dark mode implemented.

TurturOk committed:
Removing css file that was replaced by scss. Darker colors for table sections' headers.

raxsade reviewed (changes requested):
SCSS is acceptable.

I don't think the white flash can be fixed with adjusting rendering - it just takes time to load up javascript and read from localStorage.

You can replace the current logo with the one with a white outline: https://drive.google.com/file/d/1zVzCCxZutijtmjBjAbxLDMlATshLDCMO/view?usp=sharing

When filling the survey, there's a menu at the top right corner with a fixed position and with narrower screens the dark mode icon gets covered by it. And going to even narrower screens, the dark mode icon gets aligned to the left under the title, which looks out of place.

Also, the consistency of class names got fixed in the master branch after you created the pull request, so there are some conflicts that need to be resolved in this branch before the pull request can be accepted.

TurturOk committed:
- Dark mode switch moved from Header to Menu component and floats along with user name box.
- Main logo replaced by one with additional white borders.
- Dark mode switch got a pointer cursor.
- Dark mode switch no longer appears when printing pages.
- Dark mode classes renamed for consistency

TurturOk committed:
Main logo size reduced. Requested code cleanup.

raxsade committed:
Merge branch 'master' into pr/19

raxsade committed:
fix: bump github node version
